src/Client.php

```diff
@@ -227,18 +245,33 @@ protected function restCall($apiUrl, $verb, $guzzleParams, $apiversion='',$heade
             $JsonBody = json_decode($body);
             if (is_null($JsonBody)) {
                 if (json_last_error() === JSON_ERROR_SYNTAX) {
-				     throw new \Exception('The response is in unexpected format. The response is: ' . $JsonBody);
-			     }
-              return $body;
+                    $errorMsg = 'The response is in unexpected format. The response is: ';
+                    // populate exception details in log object
+                    $logModel-> populateErrorInfoWithMessageAndBody($errorMsg, $response);
+				    throw new \Exception($errorMsg . $JsonBody);
+			    }
+                $logModel-> populateResponseInfo($body, $response);
```

Comment:
nit: remove whitespace between -> and method name, the rest of the SDK follows this convention so it would be good if we can keep it consistant

Reply:
Done